src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/ResettlementAssessmentService.kt

```diff
@@ -126,8 +140,112 @@ class ResettlementAssessmentService(
     }
+
+    // New way - combine and send case note to DPS and Delius
+    if (sendCombinedCaseNotes) {
+      val dpsSubType = when (assessmentType) {
+        ResettlementAssessmentType.BCST2 -> DpsCaseNoteSubType.INR
+        ResettlementAssessmentType.RESETTLEMENT_PLAN -> DpsCaseNoteSubType.PRR
+      }
+
+      val groupedAssessmentsDps = processAndGroupAssessmentCaseNotes(assessmentList, true)
+      groupedAssessmentsDps.forEach {
+        caseNotesService.postBCSTCaseNoteToDps(
+          nomsId = prisonerEntity.nomsId,
+          notes = it.caseNoteText,
+          userId = it.user.userId,
+          subType = dpsSubType,
+        )
+      }
+
+      val crn = resettlementPassportDeliusApiService.getCrn(nomsId)
+
+      if (crn != null) {
+        val groupedAssessmentsDelius = processAndGroupAssessmentCaseNotes(assessmentList, false)
+        groupedAssessmentsDelius.forEach {
+          caseNotesService.postBCSTCaseNoteToDelius(
+            crn = crn,
+            prisonCode = prisonerSearchApiService.findPrisonerPersonalDetails(prisonerEntity.nomsId).prisonId,
+            notes = it.caseNoteText,
+            name = it.user.name,
+            assessmentType = assessmentType,
+          )
+        }
+      } else {
+        // TODO PSFR-1386 Add retry mechanism if we fail to send the case note
+        log.warn("Cannot send report case note to Delius as no CRN is available for prisoner ${prisonerEntity.nomsId}.")
+      }
+    }
   }
 
+  fun processAndGroupAssessmentCaseNotes(assessmentList: List<ResettlementAssessmentEntity>, limitChars: Boolean): List<UserAndCaseNote> {
+    val maxCaseNoteLength = if (limitChars) {
+      // Limit for DPS is 4000 but set this lower to account for line breaks between each pathway and the Part x of y text at the start (should be about 25 chars)
+      3950
+    } else {
+      Int.MAX_VALUE
+    }
+
+    val userToCaseNoteMap = assessmentList.map {
+      UserAndCaseNote(
+        user = User(userId = it.createdByUserId, name = it.createdBy),
+        caseNoteText = "${it.pathway.displayName}\n\n${it.caseNoteText}",
+      )
+    }.groupBy { it.user }
+
+    val caseNoteList = userToCaseNoteMap.flatMap { e ->
+      val combinedCaseNotes = splitToCharLimit(e.value.map { it.caseNoteText }, maxCaseNoteLength)
+      combinedCaseNotes.map {
+        UserAndCaseNote(
+          user = e.key,
+          caseNoteText = it,
+        )
+      }
+    }
+
+    return if (caseNoteList.size > 1) {
+      caseNoteList.mapIndexed { index, caseNote ->
+        UserAndCaseNote(
+          user = caseNote.user,
+          caseNoteText = "Part ${index + 1} of ${caseNoteList.size}\n\n${caseNote.caseNoteText}",
+        )
+      }
+    } else {
+      caseNoteList
+    }
+  }
+
+  fun splitToCharLimit(caseNotes: List<String>, maxCaseNoteLength: Int): List<String> {
+    val splitCaseNotes = mutableListOf<List<String>>()
+    var currentCaseNote = mutableListOf<String>()
```

Comment:
I think this code might be easier if current was just a string that got reassigned